src/test/java/com/wikia/webdriver/testcases/adstests/TestSlotsOasis.java

```diff
@@ -0,0 +1,29 @@
+package com.wikia.webdriver.testcases.adstests;
+
+import com.wikia.webdriver.common.core.Assertion;
+import com.wikia.webdriver.common.dataprovider.ads.AdsDataProvider;
+import com.wikia.webdriver.common.templates.TemplateNoFirstLoad;
+import com.wikia.webdriver.pageobjectsfactory.pageobject.adsbase.AdsBaseObject;
+
+import org.testng.annotations.Test;
+
+/**
+ * @ownership AdEng
+ */
+public class TestSlotsOasis extends TemplateNoFirstLoad {
+
+  @Test(
+      groups = {"TestSlotsOasis"},
+      dataProviderClass = AdsDataProvider.class,
+      dataProvider = "adsSlotsOasis"
+  )
+  public void adsSlotOasis(String slotName) {
+    String testedPage = urlBuilder.getUrlForPath("adtest", "SyntheticTests/OasisSlots");
+    AdsBaseObject ads = new AdsBaseObject(driver, testedPage);
+    ads.waitForPageLoaded();
+
+    Assertion.assertTrue(ads.isSlotOnPageLoaded(slotName));
+    ads.verifyGptIframe("wka.ent/_adtest//article", slotName, "gpt");
```

Comment:
And what is the reason you'd like to see that more than how it's right now? Can you help me with finding a solution for it? As the only thing I can came up right now is a wrapper method called `verifySlotLoadedOnPage()` placed in `AdsBaseObject` which I'm not sure if is better idea since `AdsBaseObject` is quite big anyway.